### PR TITLE
Generate types with the correct accessibility if they are part of public...

### PR DESCRIPTION
**Pull request by jmarolf**
... fields

fixes internal issue #1107929

### CODE REVIEW

**Comment by jmarolf** [comment]
@srivatsn @tmeschter @shyamnamboodiripad @heejaechang @mavasani @JohnHamby Could you take a look please?

File: src/EditorFeatures/CSharpTest/Diagnostics/GenerateType/GenerateTypeTests.cs [hunk 1]
```diff
@@ -1951,5 +1951,36 @@ public void TestWithUsingStatic2()
...
+
+        [WorkItem(1107929)]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateType)]
+        public void TestAccesiblityForPublicFields3()
+        {
+            Test(
+@"class A { public B b = new [|B|](); }",
+@"class A { public B b = new B(); public class B { public B() {}}}",
+index: 2);
+        }
```
Comment (shyamnamboodiripad) [inline src/EditorFeatures/CSharpTest/Diagnostics/GenerateType/GenerateTypeTests.cs]: Could you also test the cases where ```B``` is fully qualified / alias qualified? Would we generate in the correct location in such cases?

File: src/EditorFeatures/VisualBasicTest/Diagnostics/GenerateType/GenerateTypeTests.vb [hunk 2]
```diff
@@ -747,6 +747,34 @@ NewLines("Public Interface I \n  Sub Foo(a As X.Y.Z) \n End Interface \n Public
...
+index:=1)
+        End Sub
+
+        <WorkItem(1107929)>
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsGenerateType)>
+        Public Sub TestAccesiblityForPublicFields3()
+            Test(
+NewLines("Public Class A \n Public B As New [|B|]() \n End Class"),
+NewLines("Public Class A \n Public B As New B() \n Public Class B \n Public Sub New() \n End Sub \n End Class \n End Class"),
+index:=2)
```
Comment (shyamnamboodiripad) [inline src/EditorFeatures/VisualBasicTest/Diagnostics/GenerateType/GenerateTypeTests.vb]: We should probably also handle the ```= new B()``` case as well. Could you add test for that?

Comment (shyamnamboodiripad) [inline src/EditorFeatures/VisualBasicTest/Diagnostics/GenerateType/GenerateTypeTests.vb]: [minor] Probably also add tests for cases where the constructor has parameters, type is generic etc.

**Comment by shyamnamboodiripad** [comment]
LGTM modulo above comments.

**Pull request merged**